Update to Yocto 3.1.1 LTS (Dunfell)

BaochengSu opened the pull request:
Update to Yocto 3.1.1 LTS (Dunfell)

BaochengSu committed:
Switch to Yocto 3.1 (dunfell)

Update layers to latest LTS dunfell branch (3.1.1)

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
example: Set OEBasicHash as the default BB_SIGNATURE_HANDLER

uclibc do_package fails on sstate_report_unihash with the default
BB_SIGNATURE_HANDLER?=OEEquivHash on dunfell. To work around this,
by setting it to the OEBasicHash, which is the default value of old
Yoctos, the sstate_report_unihash does nothing so that it never fails.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
bsp: ell: Revert ell to 0.22

The ell version for dunfell is not compatible with our kernel

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
bsp: nodejs: Add python3 building support

The upstream Nodejs V8 building does not support python3.

What were done in this commit:
  - Use the py2to3 tool to do a quick automatic conversion.
    This is done by the first patch.
  - Then fix all the errors encountered during the build time and
    during the run time.
    This is done by the rest patches, most of which are formed
    by upstream related commits.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
Use https for meta-openembedded

git clone of http://git.openembedded.org/meta-openembedded fails,
switch to use https instead.

Signed-off-by: Su Baocheng <baocheng.su@siemens.com>

BaochengSu committed:
example: Update node-red to 1.2.1

Signed-off-by: Su Baocheng <baocheng.su@siemens.com>

BaochengSu committed:
example: Update node-red-contrib-mindconnect to 3.9.1

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
example: Add node-red-contrib-opcua 0.2.88

node-red-contrib-iiot-opcua has been deprecated, so replace it
with node-red-contrib-opcua.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
example: Update node-red-dashboard to 2.23.4

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
example: Update node-red-node-serialport to 0.11.0

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
example: Update node-red-contrib-modbus to 5.13.3

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
Update nodejs to v8.17.0

Also ported some CVE patches.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
bsp: nodejs: Ported openssl patches

Two patches are ported from:
 - OpenSUSE:nodejs8-8.17.0-lp152.147.1:openssl_1_1_1.patch
 - OpenSUSE:nodejs8-8.17.0-lp152.147.1:fix_build_with_openssl_1.1.1d.patch

Details could be found within the new added patches.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

jan-kiszka commented on meta-iot2000-bsp/conf/layer.conf:
```diff
@@ -9,7 +9,7 @@ BBFILE_COLLECTIONS += "iot2000-bsp"
 BBFILE_PRIORITY_iot2000-bsp = "6"
 
-LAYERSERIES_COMPAT_iot2000-bsp = "rocko sumo thud zeus"
+LAYERSERIES_COMPAT_iot2000-bsp = "rocko sumo thud zeus dunfell"
```
I wonder if we are really still backward-compatible... Anyway, not a blocker for this commit.

BaochengSu replied:
Fixed.

jan-kiszka reviewed (changes requested):
Shouldn't we better update to the latest ofono, at least the one that dunfell provides?

jan-kiszka reviewed (commented):
Uff, looks like you had a lot of with nodefs and python3...

jan-kiszka commented on meta-iot2000-example/recipes-node/node-red-contrib-iiot-opcua/node-red-contrib-iiot-opcua_3.12.1.bb:
```diff
@@ -3,7 +3,7 @@ SUMMARY = "The next evolution IoT/IIoT OPC UA toolbox package for Node-RED based
 # your responsibility to verify that the values are complete and correct.
 LICENSE = "Apache-2.0 & ISC & BSD & BSD-3-Clause & MIT"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=fb4562eee6b20e9a23a17f38a9f2527a"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=2419952b1e443ba100cb1c97b931ddab"
```
Any change in the licenses as well, given that the checksum changed?

BaochengSu replied:
They just add a new single year (2019) to the copyright line, the license stays the same.

BaochengSu replied:
However, the actual license is BSD-3-Clause, should we change the above line to 
```
LICENSE = "BSD-3-Clause"
```
so that it would be more accurate to reflect the license status?

jan-kiszka replied:
Before removing licenses, better to a grep on the code and also the dependencies we package.

BaochengSu replied:
Do we really need the license of the dependencies (i.e. node_modules)? I checked some community nodes layers, seems just showing the license of the node itself would be satisfied. For example https://github.com/imyller/meta-nodejs-contrib/tree/master/recipes-extended.

jan-kiszka replied:
We are also packaging dependencies, so this recipe should describe it all.

BTW, node-red-contrib-iiot-opcua should be replaced with node-red-contrib-opcua anyway as it's no longer maintained as OSS.

jan-kiszka commented on meta-iot2000-example/recipes-node/node-red-dashboard/node-red-dashboard_2.23.4.bb:
```diff
@@ -3,7 +3,7 @@ SUMMARY = "A set of dashboard nodes for Node-RED"
 # your responsibility to verify that the values are complete and correct.
 LICENSE = "Apache-2.0 & MIT & ISC"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=f289d543031cf6f8f1943a54d087eeac"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=088ce976a5d78ef059391f1ca042eb3d"
```
Same question here: New checksum == new licenses?

BaochengSu replied:
Similar here, copyright year is changed from 2019 to 2020. Also, should we change the guessed LICENSE to more accurate "Apache-2.0"?

jan-kiszka replied:
Same as above.

jan-kiszka commented on meta-iot2000-example/recipes-node/node-red-contrib-modbus/node-red-contrib-modbus_5.13.3.bb:
```diff
@@ -3,10 +3,12 @@ SUMMARY = "The all in one Modbus TCP and Serial contribution package for Node-RE
 # your responsibility to verify that the values are complete and correct.
 LICENSE = "BSD-3-Clause"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=38c55ac37ed977af78e113caffd06846"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=4ce3d0350364f3efff545d6ba54f61eb"
```
And here as well: New licenses?

BaochengSu replied:
Some new year range is added to copyright line, no new license.

jan-kiszka reviewed (commented):
Maybe leave a comment where the openssl patches for nodejs came from.

jan-kiszka commented:
Additionally, we should check if there is more to update:
 - ofono as pointed out already
 - kernel (as already discussed privately)
 - mindconnect nodes

Then I wonder if all our bbappends are still relevant and could be dropped - or if we miss some updates. Example:
 - busybox_1.31.%.bbappend
 - e2fsprogs_%.bbappend
 - nativesdk-mingw-w64-runtime_3.%.bbappend
 - mraa_%.bbappend
 - tcf-agent_git.bbappend

BaochengSu commented:
> Uff, looks like you had a lot of with nodefs and python3...

Yes, I will leave some comments on the commit messge to show how the work has been done so that it more easier to understand.

BaochengSu commented:
> Shouldn't we better update to the latest ofono, at least the one that dunfell provides?

I remember that the latest ofono from dunfell doesn't pass build for us. Sorry that I've already forgotten the exact build errors. Anyway, this change was made at the very early stage of the work, so I think it would be worth to retry the latest version now.

BaochengSu commented:
> Maybe leave a comment where the openssl patches for nodejs came from.

Sure, I will leave the comments in the commit messge.

jan-kiszka commented:
> > Shouldn't we better update to the latest ofono, at least the one that dunfell provides?
> 
> I remember that the latest ofono from dunfell doesn't pass build for us. Sorry that I've already forgotten the exact build errors. Anyway, this change was made at the very early stage of the work, so I think it would be worth to retry the latest version now.

jan-kiszka commented:
Sorry, fat-fingered "close".

BaochengSu committed:
bsp: linux-cip: Update to latest 4.4.238-cip50 and 4.4.235-cip49-rt31

The patch queue is rebased meanwhile the defconfig is upated.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
bsp: linux-cip: Remove duplicated items from iot2000-example.cfg

There are two of 'CONFIG_RTL8XXXU=m' in iot2000-example.cfg, remove
one to avoid kernel config error.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu commented:
> > Shouldn't we better update to the latest ofono, at least the one that dunfell provides?
> 
> I remember that the latest ofono from dunfell doesn't pass build for us. Sorry that I've already forgotten the exact build errors. Anyway, this change was made at the very early stage of the work, so I think it would be worth to retry the latest version now.

Just build w/ the ofono version of dunfell, no build error. So that particular commit would be dropped.

BaochengSu committed:
example: Update ofono to dunfell version (1.31)

Update ofono version from 1.26 to dunfell version (1.31)

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu commented:
> Additionally, we should check if there is more to update:
> 
> * ofono as pointed out already
> * kernel (as already discussed privately)
> * mindconnect nodes
> 
> Then I wonder if all our bbappends are still relevant and could be dropped - or if we miss some updates. Example:
> 
> * busybox_1.31.%.bbappend
> * e2fsprogs_%.bbappend
> * nativesdk-mingw-w64-runtime_3.%.bbappend
> * mraa_%.bbappend
> * tcf-agent_git.bbappend

Regarding mindconnect nodes, do you mean `node-red-contrib-mindconnect`? It's already been updated to the latest version in this PR.

BaochengSu committed:
example: mingw: Remove nativesdk-mingw-w64 bbapend

This bbapend file has been outdated for a while, so remove it.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu committed:
example: Remove e2fsprogs bbapend

move e2fsprogs-resize2fs from e2fsprogs' bbapend file to
IMAGE_INSTALL.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

jan-kiszka commented:
Yes - oversaw that, perfect.

BaochengSu committed:
example: Remove redundant file tcf-agent.init

This file does already exist in upstream, so remove it from our
overwritten layer.

Signed-off-by: Baocheng Su <baocheng.su@siemens.com>

BaochengSu commented:
> > > Shouldn't we better update to the latest ofono, at least the one that dunfell provides?
> > 
> > 
> > I remember that the latest ofono from dunfell doesn't pass build for us. Sorry that I've already forgotten the exact build errors. Anyway, this change was made at the very early stage of the work, so I think it would be worth to retry the latest version now.

The version from dunfell is pulled in, which is 1.31.

BaochengSu commented:
> > > > Shouldn't we better update to the latest ofono, at least the one that dunfell provides?
> > > 
> > > 
> > > I remember that the latest ofono from dunfell doesn't pass build for us. Sorry that I've already forgotten the exact build errors. Anyway, this change was made at the very early stage of the work, so I think it would be worth to retry the latest version now.
> 
> The version from dunfell is pulled in, which is 1.31.

jan-kiszka reviewed (approved):
In addition, I will check if we can drop the tcf-agent bbappend.
